processor/src/decorators/mod.rs

```diff
@@ -278,6 +286,138 @@ where
 
         Ok(())
     }
+
+    /// Pushes the `(depth, index, value)` triplet of a Sparse Merkle tree with the provided
+    /// `root`, and with a leaf indexed by `key`.
+    ///
+    /// The operand stack is expected to be arranged as follows:
+    /// - key, 4 elements.
+    /// - root of the Sparse Merkle tree, 4 elements.
+    ///
+    /// After a successful operation, the advice stack will look as follows:
+    /// - depth.
+    /// - index.
+    /// - leaf value, 4 elements.
+    /// - node value, 4 elements.
+    ///
+    /// The `node value` will be the hash in domain for the depth, defined as: `H_d(key, value)`.
+    /// For more information, check [vm_core::crypto::hash::Rpo256::merge_in_domain].
+    ///
+    /// If the key/value pair isn't inserted to the tree, the `node value` will be set to
+    /// `[0, 0, 0, 0]`.
+    ///
+    /// # Errors
+    /// Will return an error if:
+    /// - The provided Merkle root doesn't exist on the advice provider
+    /// - The provided key doesn't map to a word value on the advice map
+    fn inject_smtget(&mut self) -> Result<(), ExecutionError> {
```

Comment:
I think this is quite a bit more complicated than it needs to be, and assuming I understood this correctly, I'm not sure it will handle the empty leaf case in the right way. Specifically:

1. We probably don't need to try each separate tier - we can just add a method to the `MerkleStore` to retrieve the depth for a given leaf (i.e., the `get_leaf_depth()` function I suggested). This will need to be carried through the `AdviceProvider` but that should result in much more efficient and simpler code.
2. If we are returning `[ZERO; 4]` for empty nodes, how will we be able to verify the path to that node? Seems like `mtree_get` instruction will fails for this case.
3. The advice map will not contain the map for `key |-> value` - so, we can't get the value in the way you have on lines 321 - 324. What we can get from the advice map is the pre-image of the node - i.e., `node |-> (remaining_key, value)` - but I actually don't think we need to get it here (we need to get it later in MASM code).

Reply:
1. the current implementation is aiming to work with the latest version of `MerkleStore`. When we land `get_leaf_depth`, we can make an internal change on how the injector is implemented, and call it.
2. Updated on last commit; returning node value. The MASM will convert it to zeroes, if the value is absent
3. Updated on last commit; the mapping is now `node -> value` as the remaining key is computed inside `smt::get`